inference-engine/src/transformations/src/transformations/op_conversions/fq_decomposition.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/op_conversions/fq_decomposition.hpp"
+
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/opsets/opset5.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/builder/autobroadcast.hpp>
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::FakeQuantizeDecomposition, "FakeQuantizeDecomposition", 0);
+
+bool isNeedToDecomposFQ(std::shared_ptr<ngraph::opset1::FakeQuantize> fq) {
+    if (fq->get_input_node_shared_ptr(0)->get_shape().size() > 5)
+        return true;
+
+    for (size_t i = 1; i < fq->get_input_size(); i++) {
+        std::shared_ptr<ngraph::Node> node = fq->get_input_node_shared_ptr(i);
+        if (!std::dynamic_pointer_cast<const ngraph::opset1::Constant>(node) ||
+            node->get_shape().size() > 5)
+            return true;
+    }
+
+    for (size_t i = 1; i < fq->get_input_size(); i++) {
+        auto node = fq->get_input_node_shared_ptr(i);
+
+        size_t count_not_unit_axis = 0;
+        auto shape = node->get_shape();
+        if (ngraph::shape_size(shape) != 1) {
+            size_t not_unit_axis = 0;
+            for (size_t i = 0; i < shape.size(); i++) {
+                if (shape[i] > 1) {
+                    not_unit_axis = i;
+                    count_not_unit_axis++;
+                }
+            }
+            if (count_not_unit_axis > 1 || not_unit_axis > 1)
+                return true;
+        }
```

Comment:
Are you trying to identify value on spatial dimension? Looks like it's too complex, can we simplify? for example:
``` cpp
for (size_t i = 0; i < shape.size(); i++) {
    if (i == 1ul) {
        continue;
    }
    if (shape[i] > 1ul) {
        return true;
    }
}
```